### PR TITLE
fix(NcModal): Adjust modal header name and actions

### PR DESCRIPTION
**Pull request by susnux**
### ☑️ Resolves

- Fix size and margin of header name to work with new server styles on Nextcloud 30+
- Fix focus-visible, hover and open color of header actions on bright color theme

@artonge this will fix the currently failing visual regression tests of Activity.

### 🖼️ Screenshots

🏚️ Before | 🏡 After
---|---
Screenshot_20240531_103246|Screenshot_20240531_103040
Screenshot_20240531_103251|Screenshot_20240531_103009
Screenshot_20240531_103255|Screenshot_20240531_103014

### 🏁 Checklist

- [ ] ⛑️ Tests are included or are not applicable
- [x] 📘 Component documentation has been extended, updated or is not applicable
- [x] 3️⃣ Backport to `next` requested with a Vue 3 upgrade

### CODE REVIEW

**Review by skjnldsv — APPROVED** [review]

**Review by ShGKme — APPROVED** [review]

**Review by marcoambrosini — APPROVED** [review]

**Comment by Antreesy** [comment]
SPDX header is missing 🥲

File: src/components/NcModal/NcModal.vue [hunk 1]
```diff
@@ -214,10 +217,10 @@ export default {
 			tabindex="-1">
 			<!-- Header -->
 			<transition name="fade-visibility" appear>
-				<div class="modal-header">
+				<div class="modal-header" data-theme-dark>
```
Comment (skjnldsv) [inline src/components/NcModal/NcModal.vue:220]: This doesn't seem to work if a theme is enforced :warning:

Comment (skjnldsv) [reply src/components/NcModal/NcModal.vue:220]: That requires a fix on server I think, to ensure we at least load the dark theme with data attribute if another one is enforced :thinking:

Comment (skjnldsv) [reply src/components/NcModal/NcModal.vue:220]: https://github.com/nextcloud/server/pull/46504

Comment (susnux) [reply src/components/NcModal/NcModal.vue:220]: But then talk has a problem as they are using this to have the dark "chat view" on every theme

Comment (skjnldsv) [reply src/components/NcModal/NcModal.vue:220]: Yes, that's k, this is an edge case :)
Hope the server fix suits you! :+1:

**Pull request merged**